ui/src/app/views/concept-set-list/component.tsx

```diff
@@ -0,0 +1,123 @@
+    constructor(props) {
+      super(props);
+      this.state = {
+        conceptSetsList: undefined,
+        conceptSetsLoading: true,
+        conceptDomainsList: undefined,
+        createModalOpen: false
+      };
+    }
+
+    componentDidMount() {
+      try {
+        this.getDomainInfo();
+      } finally {
+        this.loadConceptSets();
+      }
+    }
+
+    async getDomainInfo() {
+      const {namespace, id} = this.props.workspace;
+      try {
+        const resp = await conceptsApi().getDomainInfo(namespace, id);
+        this.setState({conceptDomainsList: resp.items});
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    async loadConceptSets() {
+      const {namespace, id, accessLevel} = this.props.workspace;
+      try {
+        const resp = await conceptSetsApi().getConceptSetsInWorkspace(namespace, id);
+        const conceptSets = convertToResources(resp.items, namespace, id,
+          accessLevel as unknown as WorkspaceAccessLevel, ResourceType.CONCEPT_SET);
+
+        this.setState({conceptSetsList: conceptSets, conceptSetsLoading: false});
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    render() {
+      const {conceptSetsLoading, conceptSetsList, conceptDomainsList, createModalOpen} = this.state;
+      const accessLevel = this.props.workspace.accessLevel as unknown as WorkspaceAccessLevel;
+      const writePermission = accessLevel === WorkspaceAccessLevel.OWNER ||
+          accessLevel === WorkspaceAccessLevel.WRITER;
+
+      return <React.Fragment>
+            <div style={styles.pageArea}>
+              <TooltipTrigger content={!writePermission &&
+              `Write permission required to create cohorts`} side='top'>
+                <CardButton type='small'
+                            onClick={() => this.setState({createModalOpen: true})}
+                            disabled={!writePermission}>
+                  Create a <br/>Concept Set
+                  <ClrIcon shape='plus-circle' size={21} style={{marginTop: 5}}/>
+                </CardButton>
+              </TooltipTrigger>
+              <div style={styles.resourceCardArea}>
+                {conceptSetsList && conceptSetsList.map((set: RecentResource) => {
+                  return <ResourceCard resourceCard={set} key={set.conceptSet.name}
+                                       onUpdate={() => this.loadConceptSets()}>
+                  </ResourceCard>;
+                })}
+              </div>
+              {conceptSetsLoading && <SpinnerOverlay />}
+            </div>
+
+        {createModalOpen &&
+        <CreateConceptSetModal onCreate={() => this.loadConceptSets()}
+                                        onClose={() => this.setState({createModalOpen: false})}
+                                        conceptDomainList={conceptDomainsList}
+                                        existingConceptSets={conceptSetsList}/>}
```

Comment:
well i think `conceptSetsList` should be plural like that bc it contains multiple concept sets.  I changed `conceptDomainsList` to be the singular because it's referring to one domain.